### PR TITLE
ppc64le specific changes added to prow config

### PR DESCRIPTION
**Pull request by valen-mascarenhas14**
Signed-off-by: Valen Mascarenhas [Valen.Mascarenhas@ibm.com](mailto:Valen.Mascarenhas@ibm.com)

Which issue(s) this PR fixes:

Fixes

ppc64le specific changes added to prow config to support the new changes in the upstream ci repo

### CODE REVIEW

File: prow/jobs_config/knative-extensions/kn-plugin-event.yaml [hunk 1]
```diff
@@ -49,10 +49,12 @@ jobs:
...
-        mkdir -p /root/.kube
-        cat /opt/cluster/ci-script > connect.sh
-        chmod +x connect.sh
-        ./connect.sh ${CI_JOB}
+        cat /opt/cluster/vm-script > /tmp/vm.sh
+        cat /opt/cluster/ci-script > /tmp/connect.sh
+        chmod +x /tmp/vm.sh
+        chmod +x /tmp/connect.sh
+        server_vm=$(. /tmp/vm.sh)
+        . /tmp/connect.sh ${CI_JOB} ${server_vm}
```
Comment (cardil) [inline prow/jobs_config/knative-extensions/kn-plugin-event.yaml:57]: Those executions could all be simplified like this:

```suggestion
        server_vm="$(sh /opt/cluster/vm-script)"
        source /opt/cluster/ci-script "${CI_JOB}" "${server_vm}"
```

**Review by cardil — COMMENTED** [review]
From where is this change coming from? I mean, there are some s390x clusters dedicated for upstream Knative?

File: prow/jobs_config/knative-extensions/kn-plugin-event.yaml (hunk 1 above)
Comment (valen-mascarenhas14) [reply prow/jobs_config/knative-extensions/kn-plugin-event.yaml:57]: Hey @cardil , on ppc64le. We don't have dedicated clusters for Knative upstream , we recreate the clusters for every test we run . I will incorporate these changes u mention

File: prow/jobs/generated/knative-extensions/kn-plugin-event-main.gen.yaml [hunk 2]
```diff
@@ -125,10 +125,12 @@ periodics:
       - bash
       - -c
       - |
-        mkdir -p /root/.kube
-        cat /opt/cluster/ci-script > connect.sh
-        chmod +x connect.sh
-        ./connect.sh ${CI_JOB}
+        cat /opt/cluster/vm-script > /tmp/vm.sh
```
Comment (kvmware) [inline prow/jobs/generated/knative-extensions/kn-plugin-event-main.gen.yaml:128]: Plesase simplify this bash as @cardil already pointed out across all these files.

**Comment by valen-mascarenhas14** [comment]
@cardil  Have made the necessary changes as requested by you

**Comment by kvmware** [comment]
This is touching jobs across many different teams... I would like to see a +1 from at least serving or eventing that this is not disrupting them any.

**Comment by valen-mascarenhas14** [comment]
Hey @kvmware 

These changes are specific to ppc64le and do not introduce any major modifications. Previously, we were using a hardcoded IP address for VM when recreating the cluster for testing purposes. With these changes, we've implemented a script that dynamically selects a free VM and retrieves its IP address for cluster recreation.
All other aspects will continue to function as they did before. If there are any concerns or need for further clarification, please feel free to reach out.

**Comment by valen-mascarenhas14** [comment]
@cardil @dprotaso @psschwei @kvmware  any updates ?

**Review by cardil — APPROVED** [review]
/lgtm

**Pull request merged**